crates/cac_toml/src/lib.rs

```diff
@@ -0,0 +1,478 @@
+use pest::iterators::Pair;
+use pest::Parser;
+use pest_derive::Parser;
+use std::cmp::Ordering;
+use std::collections::{BinaryHeap, HashMap};
+use std::fmt;
+use std::fs;
+use std::path::Path;
+use std::string::String;
+use toml::Value;
+
+// the grammar for context expressions written using PEST
+#[derive(Parser)]
+#[grammar_inline = r###"
+expression = { SOI ~ whitespace* ~ logical ~ whitespace* ~ EOI }
+
+logical = _{ logical_or }
+logical_or = { logical_and ~ (whitespace* ~ "||" ~ whitespace* ~ logical_and)* }
+logical_and = { comparison ~ (whitespace* ~ "&&" ~ whitespace* ~ comparison)* }
+
+comparison = { term ~ whitespace* ~ comparison_operator ~ whitespace* ~ term }
+comparison_operator = { ">=" | "<=" | "<" | ">" | "==" | "!=" }
+term = { bool_literal | string_literal | float | integer | dimension }
+
+string_literal = @{ "'" ~ char+ ~ "'" }
+bool_literal = @{ "true" | "false" }
+integer = @{ ASCII_DIGIT+ }
+float = @{ ASCII_DIGIT+ ~ "." ~ ASCII_DIGIT+ }
+char = { ASCII_ALPHANUMERIC | "." | "_" }
+dimension = @{ "$" ~ char+ }
+
+whitespace = _{ " " | "\t" | "\n" }
+"###]
+struct CACParser;
+
+#[derive(Debug)]
+struct ContextualOverride {
+    expression: String,
+    extracted_dimensions: Vec<String>,
+    overrides: Value,
+    priority: i64,
+}
+
+impl PartialEq for ContextualOverride {
+    fn eq(&self, other: &Self) -> bool {
+        (self.extracted_dimensions == other.extracted_dimensions)
+            && (self.expression == other.expression)
+    }
+}
+
+impl Eq for ContextualOverride {}
```

Comment:
nitpick: We can derive Eq on the struct itself

Reply:
@Datron - did not understand, newbie question - you mean using a macro?

Reply:
<img width="1302" alt="Screenshot 2024-09-06 at 3 06 10 PM" src="https://github.com/user-attachments/assets/40750439-33dc-4eb3-9838-7011032fc965">

I think I will need to define Eq for Value as well (which is a type I don't own)